controllers/verificationController.js

```diff
@@ -0,0 +1,60 @@
+const { sendVerificationEmail } = require("../controllers/email");
+const User = require("../models/usersSchema");
+
+const verifyUser = async (req, res, next) => {
+  try {
+    const { verificationToken } = req.params;
+
+    const user = await User.findOne({ verificationToken });
+
+    if (!user) {
+      return res.status(404).send({ message: "Not found" });
+    }
+
+    user.verify = true;
+    user.verificationToken = null;
+
+    try {
+      await user.save();
+    } catch (saveError) {
+      console.error("Error saving user:", saveError);
+      return res.status(500).send({ message: "Error updating user" });
```

Comment:
Może throw? Wtedy zostanie to przesłane do bloku kolejnego catch i obsłużone przez next

Reply:
Podoba mi się że uzywasz bloków try catch nawet do małych fragmentów. Dzięki takiemu podejściu bardziej kontrolujesz przepływ informacji do FE na temat błędów.